### PR TITLE
[ETCM-126] single actor system in JsonRPController specs

### PR DESCRIPTION
**Pull request by lemastero**
# Description

Fix [review remark](https://github.com/input-output-hk/mantis/pull/734#discussion_r505271437) use single ActorSystem in JsonRpcController*Spec and properly close it.

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthSpec.scala [hunk 1]
```diff
@@ -17,22 +19,22 @@ import io.iohk.ethereum.jsonrpc.PersonalService._
...
+import org.scalatest.flatspec.AnyFlatSpecLike
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 
 import scala.concurrent.Future
 
 // scalastyle:off magic.number
 class JsonRpcControllerEthSpec
-    extends AnyFlatSpec
-    with Matchers
+    extends TestKit(ActorSystem("JsonRpcControllerSpec_System"))
```
Comment (mmrozek) [inline src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthSpec.scala:35]: We should use different names for each spec. `JsonRpcControllerEthSpec_System`

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthTransactionSpec.scala [hunk 2]
```diff
@@ -12,21 +14,23 @@ import io.iohk.ethereum.jsonrpc.JsonSerializers.{
...
+import org.scalatest.flatspec.AnyFlatSpecLike
 import org.scalatest.matchers.should.Matchers
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 
 import scala.concurrent.Future
 
 // scalastyle:off magic.number
 class JsonRpcControllerEthTransactionSpec
-    extends AnyFlatSpec
+    extends TestKit(ActorSystem("JsonRpcControllerSpec_System"))
```
Comment (mmrozek) [inline src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthTransactionSpec.scala:31]: ditto

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerPersonalSpec.scala [hunk 3]
```diff
@@ -16,14 +18,16 @@ import org.json4s.JsonAST._
...
-import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.flatspec.AnyFlatSpecLike
 import org.scalatest.matchers.should.Matchers
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 
 import scala.concurrent.Future
 
 class JsonRpcControllerPersonalSpec
-    extends AnyFlatSpec
+    extends TestKit(ActorSystem("JsonRpcControllerSpec_System"))
```
Comment (mmrozek) [inline src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerPersonalSpec.scala:28]: ditto

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthSpec.scala (hunk 1 above)
Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthSpec.scala]: Done: https://github.com/input-output-hk/mantis/pull/751/commits/1e9b0bec3cb8eb9436425a2e362ec749d826a241

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthTransactionSpec.scala (hunk 2 above)
Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerEthTransactionSpec.scala]: Done: https://github.com/input-output-hk/mantis/pull/751/commits/1e9b0bec3cb8eb9436425a2e362ec749d826a241

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerPersonalSpec.scala (hunk 3 above)
Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerPersonalSpec.scala]: Done: https://github.com/input-output-hk/mantis/pull/751/commits/1e9b0bec3cb8eb9436425a2e362ec749d826a241

**Review by mmrozek — APPROVED** [review]
lgtm

**Pull request merged**